### PR TITLE
adding Montsinger Trio

### PR DESCRIPTION
**Pull request by Rossman360**
## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [ x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [ x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x ] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ x] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [ ] I have added tests to cover my changes.
- [ x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/montsinger/trio/rev1.c [hunk 1]
```diff
@@ -0,0 +1,18 @@
+#include "rev1.h"
```
Comment (drashna) [inline keyboards/montsinger/trio/rev1.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/montsinger/trio/trio.c [hunk 2]
```diff
@@ -0,0 +1 @@
+#include "trio.h"
```
Comment (drashna) [inline keyboards/montsinger/trio/trio.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/montsinger/trio/trio.h [hunk 3]
```diff
@@ -0,0 +1,38 @@
...
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#pragma once
+
+#include "trio.h"
```
Comment (drashna) [inline keyboards/montsinger/trio/trio.h:18]:
```suggestion
```

File: keyboards/montsinger/trio/config.h [hunk 4]
```diff
@@ -0,0 +1,64 @@
...
+
+#include "config_common.h"
+
+/* USB Device descriptor parameter */
+#define VENDOR_ID       0x524D
+#define PRODUCT_ID      0x0002
+#define DEVICE_VER      0x0001
+#define MANUFACTURER    Montsinger
+#define PRODUCT         Trio Rev. 1
+#define DESCRIPTION     Trio of keys and/or encoders
```
Comment (drashna) [inline keyboards/montsinger/trio/config.h:28]:
```suggestion
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/montsinger/trio/rules.mk [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
+NKRO_ENABLE = no            # USB Nkey Rollover
+BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality by default
+RGBLIGHT_ENABLE = no       # Enable keyboard RGB underglow
+MIDI_ENABLE = no            # MIDI support
+UNICODE_ENABLE = no         # Unicode
+BLUETOOTH_ENABLE = no       # Enable Bluetooth with the Adafruit EZ-Key HID
+AUDIO_ENABLE = no           # Audio output on port C6
+FAUXCLICKY_ENABLE = no      # Use buzzer to emulate clicky switches
+ENCODER_ENABLE = yes
```
Comment (zvecr) [inline keyboards/montsinger/trio/rules.mk:33]:
```suggestion
# Bootloader selection
BOOTLOADER = atmel-dfu

# Build Options
#   change yes to no to disable
#
BOOTMAGIC_ENABLE = lite     # Virtual DIP switch configuration
MOUSEKEY_ENABLE = yes       # Mouse keys
EXTRAKEY_ENABLE = yes       # Audio control and System control
CONSOLE_ENABLE = no         # Console for debug
COMMAND_ENABLE = no         # Commands for debug and configuration
# Do not enable SLEEP_LED_ENABLE. it uses the same timer as BACKLIGHT_ENABLE
SLEEP_LED_ENABLE = no       # Breathing sleep LED during USB suspend
# if this doesn't work, see here: https://github.com/tmk/tmk_keyboard/wiki/FAQ#nkro-doesnt-work
NKRO_ENABLE = no            # USB Nkey Rollover
BACKLIGHT_ENABLE = no       # Enable keyboard backlight functionality
RGBLIGHT_ENABLE = no        # Enable keyboard RGB underglow
BLUETOOTH_ENABLE = no       # Enable Bluetooth
AUDIO_ENABLE = no           # Audio output
ENCODER_ENABLE = yes
```

File: keyboards/montsinger/trio/rev1.c [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+    rgblight_enable(); // Enable RGB by default
+    rgblight_sethsv(0, 255, 255);  // Set default HSV - red hue, full saturation, full brightness
+#ifdef RGBLIGHT_ANIMATIONS
+    rgblight_mode(RGBLIGHT_MODE_RAINBOW_SWIRL + 2); // set to RGB_RAINBOW_SWIRL by default
+#endif
+#endif
+
+    eeconfig_update_kb(0);
+    eeconfig_init_user();
+}
```
Comment (zvecr) [inline keyboards/montsinger/trio/rev1.c:31]:
```suggestion
```

**Review by zvecr — CHANGES_REQUESTED** [review]

**Review by drashna — APPROVED** [review]

**Comment by Rossman360** [comment]
what the hellll

**Comment by Rossman360** [comment]
well, that was silly. Hopefully it's okay that all the changes so far from this PR are now under https://github.com/qmk/qmk_firmware/pull/12268 instead

**Pull request closed without merging**